crates/config/src/madara.rs

```diff
@@ -56,93 +55,57 @@ pub enum MadaraPresetType {
     Custom,
 }
 
-#[derive(Clone)]
+/// Madara preset configuration (Type and optional path)
+#[derive(Debug, Clone, Args)]
 pub struct MadaraPreset {
+    #[arg(short, long)]
     pub preset_type: MadaraPresetType,
+    #[arg(long)]
     pub path: Option<String>,
 }
 
-pub struct MadaraRunnerConfigSequencer {
-    pub base_path: Option<String>,
-    pub preset: Option<MadaraPreset>,
-    // l1_endpoint has to be set as environmental variable
-}
-
+/// Enum representing all available Madara Runner configurations
+#[derive(Debug, Clone)]
 pub enum MadaraRunnerParams {
     Devnet(MadaraRunnerConfigDevnet),
     Sequencer(MadaraRunnerConfigSequencer),
     FullNode(MadaraRunnerConfigFullNode),
 }
 
-impl Default for MadaraRunnerParams {
-    fn default() -> Self {
-        Self::Devnet(MadaraRunnerConfigDevnet::default())
-    }
-}
-
+/// Configuration for Madara Runner mode (with selected parameters)
+#[derive(Debug, Clone, Args)]
 pub struct MadaraRunnerConfigMode {
+    #[arg(short, long, default_value = "Madara")]
     pub name: String,
-    pub mode: Option<MadaraMode>,
+    #[clap(subcommand)]
```

Comment:
This is related to what you asked today @HermanObst. Adding the mode as part of the command, allows us to get rid of MadaraMode and just create the right arguments but it requires an extra param in the CLI. So, the user cannot select the mode but add it as part of the original command.